tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SQS/ContextPropagation.cs

```diff
@@ -84,5 +89,62 @@ public void Set(StringBuilder carrier, string key, string value)
                 carrier.AppendFormat("\"{0}\":\"{1}\",", key, value);
             }
         }
+
+        /// <summary>
+        /// The adapter to use to append stuff to a string builder where a json is being built
+        /// </summary>
+        private readonly struct StringBuilderJsonAdapter : IBinaryHeadersCollection
+        {
+            private readonly StringBuilder _carrier;
+
+            public StringBuilderJsonAdapter(StringBuilder carrier)
+            {
+                _carrier = carrier;
+            }
+
+            public byte[] TryGetLastBytes(string name)
+            {
+                throw new NotImplementedException("this adapter can only be use to write to a StringBuilder, not to read data");
+            }
+
+            public void Add(string key, byte[] value)
+            {
+                _carrier.AppendFormat("\"{0}\":\"{1}\",", key, Convert.ToBase64String(value));
+            }
+        }
+
+        /// <summary>
+        /// The adapter to use to read attributes packed in a json string under the _datadog key
+        /// </summary>
+        public readonly struct MessageAttributesAdapter : IBinaryHeadersCollection
```

Comment:
nit: this type is only used _outside_ this class, so it probably should live elsewhere rather than nested

Reply:
I put it there because I wanted it to be next to the inject side of it. I'll do something nicer.